src/lib/components/fileUploader/fileUploader.js

```diff
@@ -0,0 +1,234 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import Uppy from "@uppy/core";
+import AwsS3 from "@uppy/aws-s3";
+import Webcam from "@uppy/webcam";
+import { DashboardModal, Dashboard } from "@uppy/react";
+import { checkValidations } from "../../utils/validation";
+import {
+  componentPropTypes,
+  defaultComponentPropTypes,
+} from "../../utils/componentPropTypes";
+
+import styles from "./fileUploader.module.scss";
+
+class FileUploader extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalOpen: false,
+      validationErrors: "",
+    };
+    this.uppy = Uppy({
+      restrictions: {
+        maxNumberOfFiles: props.maxNumberOfFiles,
+        allowMultipleUploads: props.allowMultipleUploads,
+        autoProceed: props.autoProceed,
+        maxFileSize: props.maxFileSize || undefined,
+        minNumberOfFiles: props.minNumberOfFiles || undefined,
+        allowedFileTypes: props.allowedFileTypes || undefined,
+      },
+      onBeforeUpload: files => {
+        return Object.keys(files).reduce((accum, key) => {
+          if (!files[key].meta.isUploaded) {
+            accum[key] = files[key];
+          }
+          return accum;
+        }, {});
+      },
+    });
+    this.uppy.use(Webcam);
+    this.uppy.use(AwsS3, {
+      companionUrl: props.companionUrl,
+      // serversHeader I couldn't get them into the server
+      serverHeaders: {
+        folderName: props.folderInS3Name,
+      },
+    });
+    this.uppy.on("upload-success", (file, data) =>
+      this.handleUploadFiles(file, data)
+    );
+    this.uppy.on("cancel-all", () => this.handleUploadFiles());
+    this.uppy.on("file-added", file => {
+      if (file.meta.isUploaded) {
+        this.uppy.setFileState(file.id, {
+          progress: { uploadComplete: true, uploadStarted: false },
+        });
+      }
+    });
+
+    const filesUploadedArray = this.getFilesUploadedAsArray(this.props);
+    if (filesUploadedArray.length) {
+      filesUploadedArray.forEach(file => {
+        this.uppy.addFile({
+          name: file,
```

Comment:
The `name: file` I don't really understand, for me file is usually an object and to get the name I would write `file.name` but you already have the name from the function `getFilesUploadedAsArray` maybe its better to write `forEach(fileName => ...{name: fileName, ...` or just `forEach(name => ... { name, type: ''...`

Reply:
That's true...fixed